modules/image_processors/LineBoundingBoxDetector.cpp

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file   LineBoundingBoxDetector.h
+ * @brief  ライントレース用の画像処理クラス
+ * @author takuchi17 miyahara046 HaruArima08
+ */
+#include "LineBoundingBoxDetector.h"
+
+// デフォルトコンストラクタ
+LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& _lowerHSV,
+                                                 const cv::Scalar& _upperHSV)
+  : lowerHSV(_lowerHSV), upperHSV(_upperHSV)
+{
+  roi = cv::Rect(50, 240, 540, 240);
+  resolution = cv::Size(640, 480);
+  validateParameters();
+}
+
+// ROIを指定するオーバーロードコンストラクタ
+LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& lowerHSV,
+                                                 const cv::Scalar& upperHSV, const cv::Rect& roi)
+  : lowerHSV(lowerHSV), upperHSV(upperHSV), roi(roi)
+{
+  resolution = cv::Size(640, 480);
+  validateParameters();
+}
+
+// ROIと解像度を指定するオーバーロードコンストラクタ
+LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& _lowerHSV,
+                                                 const cv::Scalar& _upperHSV, const cv::Rect& _roi,
+                                                 const cv::Size& _resolution)
+  : lowerHSV(_lowerHSV), upperHSV(_upperHSV), roi(_roi), resolution(_resolution)
+{
+  validateParameters();
+}
```

Comment:
こんな感じで書いた方がコード量少ない？
再利用性とかもよさそうかも

```suggestion
// 解像度まで指定するメインコンストラクタ
LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& _lowerHSV,
                                                 const cv::Scalar& _upperHSV,
                                                 const cv::Rect& _roi,
                                                 const cv::Size& _resolution)
  : lowerHSV(_lowerHSV), upperHSV(_upperHSV), roi(_roi), resolution(_resolution)
{
  validateParameters();
}

// ROIまで指定するコンストラクタ
LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& _lowerHSV,
                                                 const cv::Scalar& _upperHSV,
                                                 const cv::Rect& _roi)
  : LineBoundingBoxDetector(_lowerHSV, _upperHSV, _roi, cv::Size(640, 480))
{}

// 最小限のコンストラクタ
LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& _lowerHSV,
                                                 const cv::Scalar& _upperHSV)
  : LineBoundingBoxDetector(_lowerHSV, _upperHSV, cv::Rect(50, 240, 540, 240), cv::Size(640, 480))
{}
```